django/db/models/lookups.py

```diff
@@ -282,6 +282,9 @@ def process_rhs(self, compiler, connection):
             return super(In, self).process_rhs(compiler, connection)
 
     def get_rhs_op(self, connection, rhs):
+        if rhs == "()":
```

Comment:
This part doesn't feel right to me. This means that we didn't successfully catch all code paths through the `process_rhs` method. "direct values" look like they're successfully caught, it's just when the rhs is a queryset (or multiple-value-producing-expression) that we need to rely on the get_rhs_op.

Unfortunately, I can't think of a better way of handling this at the moment. Maybe @akaariai has an idea?

If this part goes ahead though, we should clearly document why it's necessary so as not to confuse future us.